explainaboard/visualizers/draw_charts.py

```diff
@@ -0,0 +1,265 @@
+from __future__ import annotations
+
+import argparse
+import json
+import os
+
+from matplotlib import pyplot as plt
+import numpy as np
+
+from explainaboard.analysis.analyses import (
+    AnalysisResult,
+    BucketAnalysisResult,
+    ComboCountAnalysisResult,
+)
+from explainaboard.analysis.performance import Performance
+from explainaboard.info import SysOutputInfo
+from explainaboard.utils.logging import progress
+from explainaboard.utils.typing_utils import unwrap
+from explainaboard.visualizers.bar_chart import make_bar_chart
+
+
+def plot_combo_counts(
+    combo_results: list[ComboCountAnalysisResult], output_dir: str, sys_names: list[str]
+) -> None:
+    """
+    Plot combo count results
+    :param combo_results: The analyses to write out
+    :param output_dir: The directory to write to
+    :param sys_names: The names of the systems
+    """
+
+    # get feature maps
+    if any([x.features != combo_results[0].features for x in combo_results]):
+        raise ValueError(
+            f'all combo_results must have the same features, but got '
+            f'{[x.features for x in combo_results]}'
+        )
+    feature_names = combo_results[0].features
+    feature_maps: list[dict[str, int]] = [dict() for _ in feature_names]
+    for combo_result in combo_results:
+        for feats, count in unwrap(combo_result.combo_counts):
+            for feat, featmap in zip(feats, feature_maps):
+                featmap[feat] = featmap.get(feat, 0) + count
+    # sort in descending order of frequency
+    sorted_names = [
+        [v[0] for v in sorted(x.items(), key=lambda y: -y[1])] for x in feature_maps
+    ]
+    first_set = set(sorted_names[0])
+    if all([set(x) == first_set for x in sorted_names]):
+        sorted_names = [sorted_names[0] for _ in sorted_names]
```

Comment:
The intention of this is somewhat unclear to me: why do we need to apply this restriction? and what is the reason to choose the first list of names rather than considering the whole frequency?

Reply:
The reason to apply this restriction is because when creating a confusion matrix it's easier to read one that has the columns and rows in the same order, i.e.:

```
                positive   negative
positive
negative
```

is easier to read than

```
                negative   positive
positive
negative
```

There was no particular reason for using the first element (other than it being easier), so I changed to use the total instead.